sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/pipeline/HttpPipelineBuilder.java

```diff
@@ -65,7 +66,11 @@ public HttpPipeline build() {
         if (httpClient != null) {
             client = httpClient;
         } else {
-            client = HttpClient.createDefault();
+            if (Configuration.getGlobalConfiguration().get("ENABLE_HTTP_CLIENT_SHARING", Boolean.TRUE)) {
+                client = HttpClient.getSharedInstance();
+            } else {
+                client = HttpClient.getNewInstance();
+            }
```

Comment:
Should we just us the shared instance always here and provide documentation on how to use your own HttpClient?

Reply:
Based on our offline conversation, we'll keep this code as is for now.